torch_geometric/loader/neighbor_loader.py

```diff
@@ -121,109 +209,78 @@ class NeighborLoader(torch.utils.data.DataLoader):
     def __init__(
         self,
         data: Union[Data, HeteroData],
-        num_neighbors: Union[List[int], Dict[EdgeType, List[int]]],
-        input_nodes: Union[Optional[Tensor], NodeType,
-                           Tuple[NodeType, Optional[Tensor]]] = None,
+        num_neighbors: NumNeighbors,
+        input_nodes: InputNodes = None,
         replace: bool = False,
         directed: bool = True,
         transform: Callable = None,
         **kwargs,
     ):
-        if kwargs.get('num_workers', 0) > 0:
-            torch.multiprocessing.set_sharing_strategy('file_system')
-
-        if 'collate_fn' in kwargs:
-            del kwargs['collate_fn']
         if 'dataset' in kwargs:
             del kwargs['dataset']
+        if 'collate_fn' in kwargs:
```

Comment:
Sorry to ask in a kind of random place, just trying to track down the MR which added these - I don't understand why it is required

Reply:
This is required for PyTorch Lightning as it will try to re-create a given `DataLoader` instance for every GPU process (by calling its `__init__` function). As such, `collate_fn` (an attribute of the `DataLoader`) is now present in `kwargs`.